background/providers/kat.js

```diff
@@ -0,0 +1,103 @@
+var https = require('https');
+var zlib = require('zlib');
+var path = require('path');
+var nconf = require('nconf');
+var mongoose = require('mongoose');
+var Category = require('../../app/models/Category.js');
+var Torrent = require('../../app/models/Torrent.js');
+
+nconf.use('memory');
+nconf.argv().env('__').file({
+    file: path.resolve(__dirname + '/../../config.json')
+});
+
+if(nconf.get('database:mongodb:enabled')){
+    mongoose.connect('mongodb://' + nconf.get('database:mongodb:host') + ':' + nconf.get('database:mongodb:port') + '/' + nconf.get('database:mongodb:collection'), function(err){
+        if(err){ console.log('Cannot connect to mongodb, please check your config.json'); process.exit(1); }
+    });
+} else {
+    console.log('No database is enabled, please check your config.json'); process.exit(1);
+}
+
+var duration = undefined;
+
+switch(nconf.get('providers:kat:config:duration')) {
+    case '@hourly':
+        duration = 60000;
+        break;
+    case '@daily':
+        duration = 1440000;
+        break;
+    default:
+        duration = Number(nconf.get('providers:kat:config:duration'));
+}
+
+if(!isNaN(duration)) {
+    setInterval(function () {
+        https.get('https://kat.cr/api/get_dump/hourly/?userhash=' + nconf.get('providers:kat:config:apiKey'), function (res) {
+            // Kat
+            // torrent_info_hash|torrent_name|torrent_category|torrent_info_url|torrent_download_url|size|category_id|files_count|seeders|leechers|upload_date|verified
+            if (res.headers['content-type'] === 'application/x-gzip') {
+                // pipe the response into the gunzip to decompress
+                var gunzip = zlib.createGunzip();
+                res.pipe(gunzip);
+
+                gunzip.on('data', function (data) {
+                    var lines = data.toString().split(/\r?\n/);
+                    lines.forEach(function (line) {
+                        line = line.split('|');
+                        console.log('Processing ' + line[1]);
+                        Category.findOne({
```

Comment:
I think we should try and add a function to the "generic" provider that will then be able to be extended using node's new Classes in ES6.